src/+otp/RHS.m

```diff
@@ -48,5 +118,99 @@
                 'Vectorized', obj.Vectorized, ...
                 varargin{:});
         end
+
+    end
+    
+    methods (Access = private)
+        function mat = prop2Matrix(~, p)
+            if isa(p, 'function_handle')
+                mat = [];
+            else
+                mat = p;
+            end
+        end
+        
+        function fun = prop2Function(~, p)
+            if isa(p, 'function_handle') || isempty(p)
+                fun = p;
+            else
+                fun = @(varargin) p;
+            end
+        end
+        
+        function newRHS = applyOp(obj1, obj2, op, dOpLeft, dOpRight, dOpBoth)
+            if isa(obj1, 'function_handle')
+                obj1 = otp.RHS(obj1);
+            elseif isa(obj2, 'function_handle')
+                obj2 = otp.RHS(obj2);
+            end
+            
+            if isa(obj1, 'otp.RHS')
+                primaryRHS = obj1;
+                if isa(obj2, 'otp.RHS')
+                    f = otp.RHS.mergeProp(obj1.F, obj2.F, op);
+                    merge = @(p) otp.RHS.mergeProp(obj1.(p), obj2.(p), dOpBoth);
+                    
+                    if strcmp(obj1.Vectorized, obj2.Vectorized)
+                        vectorized = obj1.Vectorized;
+                    end
+                else
+                    f = otp.RHS.mergeProp(obj1.F, obj2, op);
+                    merge = @(p) otp.RHS.mergeProp(obj1.(p), obj2, dOpLeft);
+                    vectorized = obj1.Vectorized;
+                end
+            else
+                primaryRHS = obj2;
+                f = otp.RHS.mergeProp(obj1, obj2.F, op);
+                merge = @(p) otp.RHS.mergeProp(obj1, obj2.(p), dOpRight);
+                vectorized = obj2.Vectorized;
+            end
+            
+            % Events and NonNegative practically cannot be supported and are
+            % always unset.
+            
+            % JPattern is problematic to compute for division operators due to
+            % singular patterns
+            
+            % Mass matrices introduce several difficulties. When singular, it
+            % makes it infeasible to update InitialSlope, and therefore, it is
+            % always unset. To avoid issues with two RHS' having different mass
+            % matrices, only the primary RHS is used.
+            
+            newRHS = otp.RHS(f, ...
```

Comment:
Mass matrices are a big difficulty right now it seems like.

Reply:
I don't think there is much more that can be done outside of warnings for incompatible mass matrices.

Reply:
I agree. That is fine